src/main/java/io/kestra/plugin/redis/TriggerList.java

```diff
@@ -0,0 +1,113 @@
+package io.kestra.plugin.redis;
+
+import io.kestra.core.models.annotations.Example;
+import io.kestra.core.models.annotations.Plugin;
+import io.kestra.core.models.annotations.PluginProperty;
+import io.kestra.core.models.conditions.ConditionContext;
+import io.kestra.core.models.executions.Execution;
+import io.kestra.core.models.executions.ExecutionTrigger;
+import io.kestra.core.models.flows.State;
+import io.kestra.core.models.triggers.AbstractTrigger;
+import io.kestra.core.models.triggers.PollingTriggerInterface;
+import io.kestra.core.models.triggers.TriggerContext;
+import io.kestra.core.models.triggers.TriggerOutput;
+import io.kestra.core.runners.RunContext;
+import io.kestra.core.utils.IdUtils;
+import io.kestra.plugin.redis.services.SerdeType;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+import org.slf4j.Logger;
+
+import javax.validation.constraints.NotNull;
+import java.time.Duration;
+import java.util.Optional;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+    title = "Wait for key of type list in Redis database"
+)
+@Plugin(
+    examples = {
+        @Example(
+            code = {
+                "id: watch",
```

Comment:
Prefer a full example like [here](https://kestra.io/plugins/plugin-azure/triggers/storage.blob/io.kestra.plugin.azure.storage.blob.Trigger.html) for Trigger to allow people understand how to used it for real